src/main/java/io/cdap/plugin/gcp/datastore/sink/DatastoreRecordWriter.java

```diff
@@ -116,9 +116,9 @@ public void write(NullWritable key, Entity entity) throws IOException {
         .setKey(response.getKeysList().stream().collect(MoreCollectors.onlyElement()))
         .putAllProperties(entity.getPropertiesMap())
         .build();
-      builder.addMutations(DatastoreHelper.makeInsert(fullEntity).build());
+      builder.addMutations(DatastoreHelper.makeUpsert(fullEntity).build());
```

Comment:
why does this test fail now? Was there any backwards incompatible change merged to plugin?

Reply:
This is a regression in develop branch caused by #376 where I had migrated the plugin to use `com.google.datastore.v1` library instead of `com.google.cloud.datastore`.

Reply:
I see. Would this mean users who are using insert operation on plugin version that uses `com.google.cloud.datastore` api will see pipeline failures after upgrading to version that develop is using?

Reply:
yes, user's pipeline would fail if:
- their pipeline tries to write an entity that already exists in datastore.
- they upgrade to version that develop is using (doesn't have the fix in this in PR).

Reply:
The expected behavior in this case is to overwrite the entity that already exists. [integration test](https://github.com/cdapio/cdap-integration-tests/blob/develop/integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GoogleCloudDatastoreTest.java#L214)

The old code uses  [put](https://github.com/data-integrations/google-cloud/pull/376/files#diff-9e0b36731c60e30d64ed56237bf2442cb1e23995435ccd5d84f17061776eab5dL69), which is an [upsert operation](https://googleapis.dev/java/google-cloud-datastore/latest/com/google/cloud/datastore/DatastoreBatchWriter.html#put-com.google.cloud.datastore.FullEntity-)

Reply:
hm actually only `put` is an upsert operation, [putWithDeferredIdAllocation](https://googleapis.dev/java/google-cloud-datastore/latest/com/google/cloud/datastore/DatastoreBatchWriter.html#putWithDeferredIdAllocation-com.google.cloud.datastore.FullEntity...-) is not. I'll update the PR to remove the unnecessary upsert.

Reply:
I see.